File: src/com/leetcode/Easy/UncommonWordsFromTwoSentences.java
package com.leetcode.Easy;

import java.util.*;

// 884. Uncommon Words from Two Sentences
public class UncommonWordsFromTwoSentences {

    public String[] uncommonFromSentences(String A, String B) {
        String[] a = A.split(" ");
        String[] b = B.split(" ");
        List<String> list = new ArrayList<>();
        HashMap<String, Integer> hashMap = new HashMap<>();
        for (String str : a) {
            if (hashMap.containsKey(str)) {
                int val = hashMap.get(str);
                hashMap.put(str, ++val);
            } else {
                hashMap.put(str, 1);
            }
        }
        for (String str : b) {
            if (hashMap.containsKey(str)) {
                int val = hashMap.get(str);
                hashMap.put(str, ++val);
            } else {
                hashMap.put(str, 1);
            }
        }
        for (Map.Entry<String, Integer> entry : hashMap.entrySet()) {
            if (entry.getValue() == 1) {
                list.add(entry.getKey());
            }
        }

        return list.toArray(new String[0]);
    }
}

File: JavaScript/Easy/UncommonWordsFromTwoSentences.js
/**
 * 884. Uncommon Words from Two Sentences
 * @param {string} s1
 * @param {string} s2
 * @return {string[]}
 */
const uncommonFromSentences = function (s1, s2) {
  const words1 = s1.split(' ');
  const words2 = s2.split(' ');

  const wordCount = {};
  [...words1, ...words2].forEach(word => {
    wordCount[word] = (wordCount[word] || 0) + 1;
  });

  let result = [];
  for (let word in wordCount) {
    if (wordCount[word] === 1)
      result.push(word);
  }

  return result;
};